test/curry.js

```diff
@@ -123,28 +122,38 @@ describe('curry', function() {
 });
 
 describe('curry properties', function() {
-  jsv.property('curries multiple values', funcN(4), jsv.json, jsv.json, jsv.json, jsv.json, function(f, a, b, c, d) {
-    var g = R.curry(f);
-
-    return R.all(R.equals(f(a, b, c, d)), [
-      g(a, b, c, d),
-      g(a)(b)(c)(d),
-      g(a)(b, c, d),
-      g(a, b)(c, d),
-      g(a, b, c)(d)
-    ]);
+  it('curries multiple values', function() {
+    fc.assert(fc.property(fc.func(fc.anything()), fc.anything(), fc.anything(), fc.anything(), fc.anything(), function(f, a, b, c, d) {
+      var f4 = function(a, b, c, d) {
+        return f(a, b, c, d);
+      };
+      var g = R.curry(f4);
+
+      return R.all(R.equals(f4(a, b, c, d)), [
+        g(a, b, c, d),
+        g(a)(b)(c)(d),
+        g(a)(b, c, d),
+        g(a, b)(c, d),
+        g(a, b, c)(d)
+      ]);
+    }));
   });
 
-  jsv.property('curries with placeholder', funcN(3), jsv.json, jsv.json, jsv.json, function(f, a, b, c) {
-    var _ = {'@@functional/placeholder': true, x: Math.random()};
-    var g = R.curry(f);
-
-    return R.all(R.equals(f(a, b, c)), [
-      g(_, _, c)(a, b),
-      g(a, _, c)(b),
-      g(_, b, c)(a),
-      g(a, _, _)(_, c)(b),
-      g(a, b, _)(c)
-    ]);
+  it('curries with placeholder', function() {
+    fc.assert(fc.property(fc.func(fc.anything()), fc.anything(), fc.anything(), fc.anything(), function(f, a, b, c) {
+      var _ = {'@@functional/placeholder': true, x: Math.random()};
+      var f3 = function(a, b, c) {
+        return f(a, b, c);
+      };
```

Comment:
Why do we need to generate a new function here instead of currying `f` directly?

Reply:
I initially tried to do that but actually ramda seems to use the `.length` of `f` during currying. The function `f` generated by fast-check looks something like `(..args) => {...}` so the `f.length === 0`. When you try to curry such function it does not work well.

```js
const {curry} = require('ramda');

const f = (...args) => args.reduce((a, b) => a + b, 0);
const g = (a, b, c) => a + b + c;

console.log(curry(f)(1, 2)) // =3
//console.log(curry(f)(1, 2)(3)) // does not work, as curry(f)(1, 2) evaluates to 3 and not to a function

console.log(curry(g)(1, 2)) // =function
console.log(curry(g)(1, 2)(3)) // =6
```
See https://runkit.com/dubzzz/5f8dd0a5e0a9d20019e56c1c